### ISSUE TITLE
Segmentation fault error in the image_read function

### ISSUE THREAD

**Issue by ksindwan** [ANTsX/ANTsPy#223]
**Describe the bug**
Segmentation fault error while executing  t1=ants.image_read( "testDataBrainOnly.nii.gz" )

**To Reproduce**
import ants
t1=ants.image_read( "testDataBrainOnly.nii.gz" ) (Path for the file https://github.com/ANTsXNet/brainSR/tree/master/data)

Package Version:
antspynet==0.0
antspyx==0.2.6  (tried antspyx==0.2.4 as well)
Keras==2.3.0
tensorflow==2.2.0
tensorflow-probability==0.10.1
 
RAM of instance : 256 GB

Error : 
Fatal Python error: Segmentation fault
 
Current thread 0x00007f6c175a1740 (most recent call first):
  File "/home/ubuntu/my-project-env/lib/python3.7/site-packages/ants/core/ants_image_io.py", line 434 in image_header_info
  File "/home/ubuntu/my-project-env/lib/python3.7/site-packages/ants/core/ants_image_io.py", line 517 in image_read
  File "<stdin>", line 1 in <module>

**Comment by stnava** [comment]
I can't reproduce this issue.   this is just basic I/O .... obviously, nothing would work  if we could not read and write images.   anyway, that points to an install issue on your end.  you might run these tests to make sure it's correct:

https://github.com/ANTsX/ANTsPy/blob/master/tests/run_tests.sh

the way to install is 

1.  install cmake 3.12.3 or greater 
2. make sure you have a c++ compiler
3. git clone this repo
4. cd ANTsPy
5. python3 setup.py install

see the travis or circle ci yml files otherwise.  we have some docker images if that would help - will update them today.

**Comment by ksindwan** [comment]
Output after running https://github.com/ANTsX/ANTsPy/blob/master/tests/run_tests.sh

./home/ubuntu/my-project-env/lib/python3.7/site-packages/ants/core/ants_image.py:476: RuntimeWarning:

invalid value encountered in true_divide

test_core_ants_image.py:284: RuntimeWarning:

invalid value encountered in true_divide

........./home/ubuntu/my-project-env/lib/python3.7/site-packages/ants/core/ants_image.py:487: RuntimeWarning:

overflow encountered in power

test_core_ants_image.py:303: RuntimeWarning:

overflow encountered in power

...........................
----------------------------------------------------------------------
Ran 37 tests in 17.906s

OK
........
----------------------------------------------------------------------
Ran 8 tests in 14.655s

OK
........................
----------------------------------------------------------------------
Ran 24 tests in 3.729s

OK
...
----------------------------------------------------------------------
Ran 3 tests in 0.527s

OK
..........
----------------------------------------------------------------------
Ran 10 tests in 13.739s

OK
Running learn tests
...
----------------------------------------------------------------------
Ran 3 tests in 0.638s

OK
Running registation tests
..0.17229822
0.15412956
0.14839341
.0.18319412
0.12977262
0.120083705
0.265975
0.23097424
0.17980465
....Starting affine interface registration test
..Starting long registration interface test
SyN
TVMSQC
SyNOnly
AffineFast
BOLDAffine
SyNabp
Translation
ElasticSyN
SyNBold
SyNCC
Affine
SyNBoldAff
TRSAA
DenseRigid
SyNRA
SyNAggro
SyNLessAggro
TVMSQ
Rigid
QuickRigid
BOLDRigid
Similarity
Finished long registration interface test
..All_Command_lines_OK
Using single precision for computations.
=============================================================================
The composite transform comprises the following transforms (in order): 
  1. Center of mass alignment using fixed image: 0x55e0b03788d0 and moving image: 0x55e0b03788b0 (type = Euler2DTransform)
=============================================================================
  Reading mask(s).
    Registration stage 0
      No fixed mask
      No moving mask
  number of levels = 3
  fixed image: 0x55e0b03788d0
  moving image: 0x55e0b03788b0
  fixed image: 0x55e0b01a4100
  moving image: 0x55e0b03b77d0
  fixed image: 0x55e0b01a4100
  moving image: 0x55e0b03b77d0
Dimension = 2
Number of stages = 1
Use Histogram Matching true
Winsorize image intensities false
Lower quantile = 0
Upper quantile = 1
Stage 1 State
   Image metric = Mattes
     Fixed image = Image (0x55e0b0af0960)
  RTTI typeinfo:   itk::Image<float, 2u>
  Reference Count: 2
  Modified Time: 18741548
  Debug: Off
  Object Name: 
  Observers: 
    none
  Source: (none)
  Source output name: (none)
  Release Data: Off
  Data Released: False
  Global Release Data: Off
  PipelineMTime: 0
  UpdateMTime: 18741524
  RealTimeStamp: 0 seconds 
  LargestPossibleRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  BufferedRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  RequestedRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  Spacing: [1, 1]
  Origin: [0, 0]
  Direction: 
1 0
0 1

  IndexToPointMatrix: 
1 0
0 1

  PointToIndexMatrix: 
1 0
0 1

  Inverse Direction: 
1 0
0 1

  PixelContainer: 
    ImportImageContainer (0x55e0b037cfa0)
      RTTI typeinfo:   itk::ImportImageContainer<unsigned long, float>
      Reference Count: 1
      Modified Time: 18741522
      Debug: Off
      Object Name: 
      Observers: 
        none
      Pointer: 0x55e0b051e6e0
      Container manages memory: true
      Size: 65536
      Capacity: 65536

     Moving image = Image (0x55e0b0af0b40)
  RTTI typeinfo:   itk::Image<float, 2u>
  Reference Count: 2
  Modified Time: 18741549
  Debug: Off
  Object Name: 
  Observers: 
    none
  Source: (none)
  Source output name: (none)
  Release Data: Off
  Data Released: False
  Global Release Data: Off
  PipelineMTime: 0
  UpdateMTime: 18741546
  RealTimeStamp: 0 seconds 
  LargestPossibleRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  BufferedRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  RequestedRegion: 
    Dimension: 2
    Index: [0, 0]
    Size: [256, 256]
  Spacing: [1, 1]
  Origin: [0, 0]
  Direction: 
1 0
0 1

  IndexToPointMatrix: 
1 0
0 1

  PointToIndexMatrix: 
1 0
0 1

  Inverse Direction: 
1 0
0 1

  PixelContainer: 
    ImportImageContainer (0x55e0b0346880)
      RTTI typeinfo:   itk::ImportImageContainer<unsigned long, float>
      Reference Count: 1
      Modified Time: 18741544
      Debug: Off
      Object Name: 
      Observers: 
        none
      Pointer: 0x55e0b0859b50
      Container manages memory: true
      Size: 65536
      Capacity: 65536

     Weighting = 1
     Sampling strategy = none
     Number of bins = 32
     Radius = 4
     Sampling percentage  = 1
   Transform = SyN
     Gradient step = 0.2
     Update field sigma (voxel space) = 3
     Total field sigma (voxel space) = 0
     Update field time sigma = 0
     Total field time sigma  = 0
     Number of time indices = 0
     Number of time point samples = 0
Registration using 1 total stages.

Stage 0
  iterations = 40x20x0
  convergence threshold = 1e-07
  convergence window size = 8
  number of levels = 3
  using the Mattes MI metric (number of bins = 32, weight = 1)
  preprocessing:  histogram matching the images
  using the Demons metric (weight = 1)
  preprocessing:  histogram matching the images
  using the CC metric (radius = 1, weight = 2)
  preprocessing:  histogram matching the images
  Shrink factors (level 1 out of 3): [4, 4]
  Shrink factors (level 2 out of 3): [2, 2]
  Shrink factors (level 3 out of 3): [1, 1]
  smoothing sigmas per level: [2, 1, 0]
  Using default NONE metricSamplingStrategy 

*** Running SyN registration (varianceForUpdateField = 3, varianceForTotalField = 0) ***

XXDIAGNOSTIC,Iteration,metricValue,convergenceValue,ITERATION_TIME_INDEX,SINCE_LAST
 1DIAGNOSTIC,     1, -7.094767689705e-01, inf, 2.7626e-02, 2.7626e-02, 
 1DIAGNOSTIC,     2, -7.452821731567e-01, inf, 4.4381e-02, 1.6755e-02, 
 1DIAGNOSTIC,     3, -7.804611921310e-01, inf, 6.1184e-02, 1.6803e-02, 
 1DIAGNOSTIC,     4, -8.131859302521e-01, inf, 7.9049e-02, 1.7865e-02, 
 1DIAGNOSTIC,     5, -8.384947180748e-01, inf, 9.6700e-02, 1.7651e-02, 
 1DIAGNOSTIC,     6, -8.559163808823e-01, inf, 1.1350e-01, 1.6795e-02, 
 1DIAGNOSTIC,     7, -8.726809024811e-01, inf, 1.3041e-01, 1.6912e-02, 
 1DIAGNOSTIC,     8, -8.857009410858e-01, 1.781513355672e-02, 1.4750e-01, 1.7095e-02, 
 1DIAGNOSTIC,     9, -8.931246995926e-01, 1.293500047177e-02, 1.6440e-01, 1.6895e-02, 
 1DIAGNOSTIC,    10, -9.017964005470e-01, 9.219971485436e-03, 1.8228e-01, 1.7882e-02, 
 1DIAGNOSTIC,    11, -9.069299697876e-01, 6.487615872175e-03, 2.0009e-01, 1.7810e-02, 
 1DIAGNOSTIC,    12, -9.144487380981e-01, 4.730695392936e-03, 2.1986e-01, 1.9769e-02, 
 1DIAGNOSTIC,    13, -9.213866591454e-01, 3.624365897849e-03, 2.3761e-01, 1.7748e-02, 
 1DIAGNOSTIC,    14, -9.232516288757e-01, 2.740704454482e-03, 2.5553e-01, 1.7920e-02, 
 1DIAGNOSTIC,    15, -9.255870580673e-01, 2.114957431331e-03, 2.7432e-01, 1.8795e-02, 
 1DIAGNOSTIC,    16, -9.279049634933e-01, 1.659483183175e-03, 2.9327e-01, 1.8951e-02, 
 1DIAGNOSTIC,    17, -9.284569025040e-01, 1.210379996337e-03, 3.1200e-01, 1.8725e-02, 
 1DIAGNOSTIC,    18, -9.274803996086e-01, 8.158859563991e-04, 3.3084e-01, 1.8840e-02, 
 1DIAGNOSTIC,    19, -9.293838143349e-01, 4.972732858732e-04, 3.4959e-01, 1.8748e-02, 
 1DIAGNOSTIC,    20, -9.283959865570e-01, 2.677589363884e-04, 3.6828e-01, 1.8693e-02, 
 1DIAGNOSTIC,    21, -9.283671975136e-01, 1.554130431032e-04, 3.8700e-01, 1.8718e-02, 
 1DIAGNOSTIC,    22, -9.299275875092e-01, 9.377023525303e-05, 4.0561e-01, 1.8617e-02, 
 1DIAGNOSTIC,    23, -9.291764497757e-01, 4.991313107894e-05, 4.2451e-01, 1.8896e-02, 
 1DIAGNOSTIC,    24, -9.305827617645e-01, 6.496549030999e-05, 4.4333e-01, 1.8817e-02, 
 1DIAGNOSTIC,    25, -9.297721385956e-01, 6.580467743333e-05, 4.6213e-01, 1.8801e-02, 
 1DIAGNOSTIC,    26, -9.287190437317e-01, 2.573043275333e-05, 4.8072e-01, 1.8597e-02, 
 1DIAGNOSTIC,    27, -9.303567409515e-01, 3.732599725481e-05, 4.9947e-01, 1.8743e-02, 
 1DIAGNOSTIC,    28, -9.310010671616e-01, 3.725965871126e-05, 5.1820e-01, 1.8738e-02, 
 1DIAGNOSTIC,    29, -9.336832761765e-01, 6.783840944991e-05, 5.3690e-01, 1.8691e-02, 
 1DIAGNOSTIC,    30, -9.325587749481e-01, 9.278718789574e-05, 5.5568e-01, 1.8784e-02, 
 1DIAGNOSTIC,    31, -9.325056076050e-01, 9.248047717847e-05, 5.7445e-01, 1.8772e-02, 
 1DIAGNOSTIC,    32, -9.324890375137e-01, 9.512215910945e-05, 5.9325e-01, 1.8800e-02, 
 1DIAGNOSTIC,    33, -9.322304725647e-01, 7.103845564416e-05, 6.1200e-01, 1.8745e-02, 
 1DIAGNOSTIC,    34, -9.325155019760e-01, 2.957864671771e-05, 6.3081e-01, 1.8812e-02, 
 1DIAGNOSTIC,    35, -9.332510232925e-01, 1.323451851931e-05, 6.4978e-01, 1.8967e-02, 
 1DIAGNOSTIC,    36, -9.313545227051e-01, -1.834671093093e-05, 6.6848e-01, 1.8705e-02, 
XXDIAGNOSTIC,Iteration,metricValue,convergenceValue,ITERATION_TIME_INDEX,SINCE_LAST
 1DIAGNOSTIC,     1, -7.007057666779e-01, inf, 7.1893e-01, 5.0449e-02, 
 1DIAGNOSTIC,     2, -8.271946907043e-01, inf, 7.5946e-01, 4.0534e-02, 
 1DIAGNOSTIC,     3, -8.438875675201e-01, inf, 7.9940e-01, 3.9935e-02, 
 1DIAGNOSTIC,     4, -8.541314601898e-01, inf, 8.4011e-01, 4.0713e-02, 
 1DIAGNOSTIC,     5, -8.626800775528e-01, inf, 8.8055e-01, 4.0438e-02, 
 1DIAGNOSTIC,     6, -8.703929185867e-01, inf, 9.2043e-01, 3.9878e-02, 
 1DIAGNOSTIC,     7, -8.747229576111e-01, inf, 9.6054e-01, 4.0113e-02, 
 1DIAGNOSTIC,     8, -8.804213404655e-01, 1.232060696930e-02, 1.0012e+00, 4.0703e-02, 
 1DIAGNOSTIC,     9, -8.839428424835e-01, 4.637320991606e-03, 1.0414e+00, 4.0135e-02, 
 1DIAGNOSTIC,    10, -8.869811296463e-01, 3.261899342760e-03, 1.0818e+00, 4.0432e-02, 
 1DIAGNOSTIC,    11, -8.896322250366e-01, 2.418733667582e-03, 1.1222e+00, 4.0353e-02, 
 1DIAGNOSTIC,    12, -8.894995450974e-01, 1.728755654767e-03, 1.1626e+00, 4.0456e-02, 
 1DIAGNOSTIC,    13, -8.921033143997e-01, 1.255711656995e-03, 1.2048e+00, 4.2177e-02, 
 1DIAGNOSTIC,    14, -8.914434909821e-01, 8.796535548754e-04, 1.2448e+00, 4.0051e-02, 
 1DIAGNOSTIC,    15, -8.927029371262e-01, 5.823399405926e-04, 1.2858e+00, 4.0995e-02, 
 1DIAGNOSTIC,    16, -8.941428661346e-01, 4.210456099827e-04, 1.3260e+00, 4.0136e-02, 
 1DIAGNOSTIC,    17, -8.946211934090e-01, 3.134234866593e-04, 1.3658e+00, 3.9868e-02, 
 1DIAGNOSTIC,    18, -8.948073387146e-01, 2.456833899487e-04, 1.4061e+00, 4.0207e-02, 
 1DIAGNOSTIC,    19, -8.952867984772e-01, 2.175251574954e-04, 1.4465e+00, 4.0415e-02, 
 1DIAGNOSTIC,    20, -8.954349756241e-01, 1.594779751031e-04, 1.4867e+00, 4.0215e-02, 
  Elapsed time (stage 0): 1.52835


Total elapsed time: 1.52894
... det 1
 A_solution 
0.10515 -0.994456 
0.994456 0.10515 

 det 1
 A_solution 
0.10515 -0.994456 
0.994456 0.10515 

..Using single precision for computations.
Input scalar image: 0x55e0b02a8550
Reference image: 0x55e0b0841f50
=============================================================================
The composite transform comprises the following transforms (in order): 
  1. identity (type = MatrixOffsetTransformBase)
=============================================================================
Default pixel value: 0
Interpolation type: LinearInterpolateImageFunction
Output warped image: 0x55e0aeb86180
..
----------------------------------------------------------------------
Ran 18 tests in 118.887s

OK
Running segmentation tests
.. Otsu Thresh with 3 thresholds
 Otsu Thresh with 3 thresholds
 Otsu Thresh with 3 thresholds
 Otsu Thresh with 3 thresholds
 Otsu Thresh with 3 thresholds
.....
----------------------------------------------------------------------
Ran 7 tests in 5.881s

OK
Running utils tests
........................................True
.False
False
......
----------------------------------------------------------------------
Ran 47 tests in 32.384s

OK
Running bug tests
.
----------------------------------------------------------------------
Ran 1 test in 0.588s

OK
~/ANTsPy/tests

**Comment by stnava** [comment]
ok - that's good because the tests run test_core_ants_image_io.py which tests the i/o ops.  maybe you have a corrupted image copy or something like that.  try this one attached.
[testDataBrainOnly.nii.gz](https://github.com/ANTsX/ANTsPy/files/5425058/testDataBrainOnly.nii.gz)

**Comment by ksindwan** [comment]
Same error 
Segmentation fault

**Comment by stnava** [comment]
I would recommend a clean re-installation from scratch .... it would be bizarre and nonsensical for the I/O tests to pass then fail for this image unless you had some odd system specific restrictions ... very tough to do anything about it if we can't reproduce the issue ...  none of our users have ever reported anything like this.  @ntustison  any ideas?

**Comment by ntustison** [comment]
Yeah, I have no idea.  I can't reproduce the error and nobody else has reported this same issue.  Do like Brian says and do a clean install.   Also, you can try to reproduce on a different system to which you have access.
